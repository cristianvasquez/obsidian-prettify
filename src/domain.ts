import {Options} from "mdast-util-to-markdown";


type FrontMatterOptions = {
  newHeaderTemplate?: any;
  createHeaderIfNotPresent?: boolean;
  updateHeader?: boolean;
}

type FontmatterInput = {
  tags?: string[];
  today?: moment.Moment;
};

type MarkdownPrettifierOptions = Options & FrontMatterOptions


export type { MarkdownPrettifierOptions, FontmatterInput, FrontMatterOptions};