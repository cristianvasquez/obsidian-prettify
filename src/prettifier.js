const gfm = require('remark-gfm')
const unified = require('unified')
const parse = require('remark-parse')
const images = require('remark-images')
const frontmatter = require('remark-frontmatter');
const table_writer = require('./table-writer');
const stringify = require('./stringify')
const remark = unified().use(parse).freeze()
const {NEW_HEADER_TEMPLATE} = require('./constants');
const math = require('remark-math')
const katex = require('rehype-katex')
const moment = require('moment')

function prettifier(
    content,
    {
        bullet = '-',
        emphasis = '_',
        rule = '-',
        listItemIndent = 'one', // one, mixed, tab

        createHeaderIfNotPresent: createHeaderIfNotPresent = false,
        newHeaderTemplate = NEW_HEADER_TEMPLATE,
        updateHeader = true,
        currentMoment = moment(),
    } = {}
) {
    let result = remark()
        .use(gfm)

    result = result.use(frontmatter)

    if (createHeaderIfNotPresent || updateHeader) {
        result = result.use(table_writer, {
            createHeaderIfNotPresent: createHeaderIfNotPresent,
            newHeaderTemplate: newHeaderTemplate,
            updateHeader: updateHeader,
            currentMoment: currentMoment
        })
    }

    result
        .use(math)
        .use(katex)
        .use(images)
        .use(stringify, {
            bullet: bullet,
            emphasis: emphasis,
            rule: rule,
            listItemIndent: listItemIndent
        })

    return result.process(content)
}

module.exports = prettifier
