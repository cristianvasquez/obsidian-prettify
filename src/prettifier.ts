import stringify from "./stringify";
import frontmatter from "remark-frontmatter";
import frontmatter_writer from './frontmatter-writer'
import type { MarkdownPrettifierOptions, FrontMatterOptions, FontmatterInput } from "./domain";

import {wikiLinkPlugin} from "./wikilinks";
import gfm from 'remark-gfm';
import math from 'remark-math'
import remark from "remark";


import { NEW_HEADER_TEMPLATE } from "./constants";

import moment from 'moment'

function prettifier(
  content: string,
  options: MarkdownPrettifierOptions = {    
    bullet: "-",
    fence: "~",
    emphasis: "_",
    rule: "-",
    fences: true,
    listItemIndent: "one",
    tightDefinitions:false,    
    incrementListMarker: true,
    createHeaderIfNotPresent: false,
    newHeaderTemplate: NEW_HEADER_TEMPLATE,
    updateHeader: true,
  },
  frontMatterData: FontmatterInput = { today: moment(), tags: [] }
) {

  let processor = remark()

  if (options.createHeaderIfNotPresent || options.updateHeader) {
    processor = processor
      .use(frontmatter)
      .use(frontmatter_writer, options as FrontMatterOptions, frontMatterData);
  }

  const stringifyOptions = Object.assign({}, options)

  if (!stringifyOptions.newlinesAroundHeadings) {
    stringifyOptions.join = [
      //@ts-ignore
      (left, right) => {
        if (left.type === 'heading' || right.type === 'heading') {
          return 0;
        }
        return 1;
      },
    ];
  }

  processor = processor
    .use(wikiLinkPlugin)
    .use(math)
    .use(gfm)
    // @ts-ignore
    .use(stringify, stringifyOptions);
    
  return processor.process(content);
}

export default prettifier
