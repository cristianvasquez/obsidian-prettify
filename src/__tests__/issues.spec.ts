import prettifier from "../prettifier";
import {MarkdownPrettifierOptions} from '../domain'
// @TODO
// Read with base __DIRNAME
// run_spec(__dirname, ["markdown"], { proseWrap: "always" });

describe("issues", () => {
  test("https://github.com/cristianvasquez/obsidian-prettify/issues/1", () => {
    const content = `
# issues/1

Prettifier:
- [ ] Don't break my TODO
            `;
    return prettifier(content).then((data) => {
      expect(data.contents).toMatchSnapshot();
    });
  });

  test("https://github.com/cristianvasquez/obsidian-prettify/issues/1", () => {
    const content = `
# issues/1

1. fjlkefjl
    1. feafe
2. feafea
3. feaf
    1. fea
3. fefe
            `;
    return prettifier(content).then((data) => {
      expect(data.contents).toMatchSnapshot();
    });
  });

  test("https://github.com/cristianvasquez/obsidian-prettify/issues/12", () => {
    const content = `
$$
\begin{bmatrix}
a && b \\
c && d
\end{bmatrix}
$$
            `;
    return prettifier(content).then((data) => {
      expect(data.contents).toMatchSnapshot();
    });
  });

  test("https://github.com/cristianvasquez/obsidian-prettify/issues/19", () => {
    const content = `
- X
    *   Tab
- X
    * Tab
- X
  * 2 spaces
- X
   * 3 spaces
- X
    * 4 Spaces
            `;
    return prettifier(content, {
      listItemIndent: "one",
    }).then((data) => {
      expect(data.contents).toMatchSnapshot();
    });
  });

  test("https://github.com/cristianvasquez/obsidian-prettify/issues/19", () => {
    const content = `
- X
    *   Tab
- X
    * Tab
- X
  * 2 spaces
- X
   * 3 spaces
- X
    * 4 Spaces
            `;
    return prettifier(content, {
      listItemIndent: "tab",
    }).then((data) => {
      expect(data.contents).toMatchSnapshot();
    });
  });

  test("https://github.com/cristianvasquez/obsidian-prettify/issues/35", () => {

    const content = `$$\begin{aligned}
		& y_{it} = \mu_i + x_{it}
		\\
		& x_{it} = \phi_i x_{i,t-1} + e_{it}
\end{aligned}$$`;
    return prettifier(content, {
      listItemIndent: "one",
    
    }).then((data) => {
      expect(data.contents).toMatchSnapshot();
    });
  });

  test("https://github.com/cristianvasquez/obsidian-prettify/issues/47", () => {
    const content = `$x + y$

$$x + y$$`;
    return prettifier(content, {
      listItemIndent: "one",
    }).then((data) => {
      expect(data.contents).toMatchSnapshot();
    });
  });

  test("https://github.com/cristianvasquez/obsidian-prettify/issues/49", () => {
    const content = `
\#12
#12
`;
    return prettifier(content, {
      listItemIndent: "one",
    }).then((data) => {
      expect(data.contents).toMatchSnapshot();
    });
  });

  test("https://github.com/cristianvasquez/obsidian-prettify/issues/20", () => {
    const content = `
| A title           | B title |
| -------           | ------- |
| [[ link | alias]] | B row |
`;
    return prettifier(content, {
      listItemIndent: "one",
    }).then((data) => {
      expect(data.contents).toMatchSnapshot();
    });
  });
});
