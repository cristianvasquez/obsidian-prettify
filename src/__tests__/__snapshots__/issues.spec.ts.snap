// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issues https://github.com/cristianvasquez/obsidian-prettify/issues/1 1`] = `
"# issues/1

Prettifier:

- [ ] Don't break my TODO
"
`;

exports[`issues https://github.com/cristianvasquez/obsidian-prettify/issues/1 2`] = `
"# issues/1

1. fjlkefjl
   1. feafe
2. feafea
3. feaf
   1. fea
4. fefe
"
`;

exports[`issues https://github.com/cristianvasquez/obsidian-prettify/issues/12 1`] = `
"$$
egin{bmatrix}
a && b \\\\
c && d
end{bmatrix}
$$
"
`;

exports[`issues https://github.com/cristianvasquez/obsidian-prettify/issues/19 1`] = `
"* X
  * Tab
* X
  * Tab
* X
  * 2 spaces
* X
  * 3 spaces
* X
  * 4 Spaces
"
`;

exports[`issues https://github.com/cristianvasquez/obsidian-prettify/issues/19 2`] = `
"*   X
    *   Tab
*   X
    *   Tab
*   X
    *   2 spaces
*   X
    *   3 spaces
*   X
    *   4 Spaces
"
`;

exports[`issues https://github.com/cristianvasquez/obsidian-prettify/issues/20 1`] = `
"| A title | B title |       |
| ------- | ------- | ----- |
| [[ link | alias]] | B row |
"
`;

exports[`issues https://github.com/cristianvasquez/obsidian-prettify/issues/35 1`] = `
"$$$egin{aligned}
		& y_{it} = mu_i + x_{it}
		\\\\
		& x_{it} = phi_i x_{i,t-1} + e_{it}
end{aligned}$$
$$$
"
`;

exports[`issues https://github.com/cristianvasquez/obsidian-prettify/issues/47 1`] = `
"$x + y$

$x + y$
"
`;

exports[`issues https://github.com/cristianvasquez/obsidian-prettify/issues/49 1`] = `
"#12 of 36

*
"
`;
